### PR TITLE
DM-32894: ESS: overhaul ErrorCode enum

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: python/lsst/ts/idl/enums/ESS.py [hunk 1]
```diff
@@ -25,12 +25,20 @@
...
-    NotConnected = 1
-    AlreadyConnected = 2
-    NotConfigured = 3
-    BadConfiguration = 4
-    ReadLoopFailed = 5
-    TelemetryError = 6
-    HardwareError = 7
+    Values:
+
+    * InvalidModelConfig: the configuration for one or more models
```
Comment (wvreeven) [inline python/lsst/ts/idl/enums/ESS.py:32]: I don't see this value in the list.

**Review by wvreeven — CHANGES_REQUESTED** [review]
It looks like `InvalidModelConfig` is missing from the enums.

File: python/lsst/ts/idl/enums/ESS.py (hunk 1 above)
Comment (r-owen) [reply python/lsst/ts/idl/enums/ESS.py:32]: Good catch. It turns out it wasn't useful -- invalid data client config is caught at the same time as any other invalid config and handled by the the parent class (ConfigurableCsc).

**Review by wvreeven — APPROVED** [review]
LGTM

**Pull request merged**